client/my-sites/plugins/marketplace/marketplace-domain-upsell/index.tsx

```diff
@@ -0,0 +1,209 @@
+				icon={ true }
+			>
+				{ translate( 'Domain names and their influence on SEO' ) }
+			</ExternalLink>
+			<HorizontalRule />
+		</div>
+	);
+}
+
+function getSiteNameFromURL( url ) {
+	const parts = url?.split( '.' );
+	if ( Array.isArray( parts ) && parts.length > 0 ) {
+		return parts[ 0 ];
+	}
+	return url;
+}
+
+function CalypsoWrappedMarketplaceDomainUpsell(): JSX.Element {
+	const [ selectedDomainProductUUID, setDomainProductUUID ] = useState< string >( '' );
+	const [ selectedDomain, setDomain ] = useState< DomainSuggestions.DomainSuggestion | null >(
+		null
+	);
+
+	const [ isExpandedBasketView, setIsExpandedBasketView ] = useState( false );
+	const { addProductsToCart, replaceProductsInCart, removeProductFromCart } = useShoppingCart();
+	const products = useSelector( getProductsList );
+	const previousPath = useSelector( getPreviousPath );
+	const isFetchingProducts = useSelector( isProductsListFetching );
+	const selectedSite = useSelector( getSelectedSite );
+	const { wpcom_url } = selectedSite;
+	const siteName = getSiteNameFromURL( wpcom_url );
+
+	useEffect( () => {
+		setIsExpandedBasketView( isDesktop() );
+	}, [ setIsExpandedBasketView ] );
+
+	useEffect( () => {
+		//FIXME: This code segment simulates yoast premium already being added when arriving here. To be removed when plugins page is completed.
+		if ( ! isFetchingProducts && products[ 'yoast_premium' ] ) {
+			const yoastProduct = fillInSingleCartItemAttributes(
+				{ product_slug: 'yoast_premium' },
+				products
+			);
+			replaceProductsInCart( [ yoastProduct ] );
+		}
+	}, [ isFetchingProducts, products, replaceProductsInCart ] );
+
+	const redirectToUseDomainFlow = (): void => {
+		const currentUrl = '/plugins/domain';
+		const useYourDomainUrl = `/start/new-launch/domains-launch/use-your-domain?siteSlug=${ selectedSite?.slug }&source=${ currentUrl }`;
+		page( useYourDomainUrl );
+	};
+
+	const onDomainSelect = ( suggestion: DomainSuggestions.DomainSuggestion ) => {
+		const { product_slug, domain_name } = suggestion;
+		const domainProduct = {
+			...domainRegistration( {
+				productSlug: product_slug,
+				domain: domain_name,
+				source: 'Marketplace-Yoast-Domain-Upsell',
+			} ),
+			...suggestion,
+		};
+		setDomain( domainProduct );
+
+		//First remove the previously added domain
+		new Promise< void >( ( resolve ) => {
+			if ( ! selectedDomainProductUUID ) {
+				resolve();
+			} else {
+				removeProductFromCart( selectedDomainProductUUID ).then( () => {
+					setDomainProductUUID( '' );
+					resolve();
+				} );
+			}
+		} ).then( () => {
+			//Then add the new domain
+			addProductsToCart( [ domainProduct ] ).then( ( responseCart: ResponseCart ) => {
+				const productAdded = responseCart.products.find(
+					( { product_slug: added_product_slug } ) => added_product_slug === product_slug
+				);
+				setDomainProductUUID( productAdded?.uuid ?? '' );
+			} );
+		} );
+	};
+
+	return (
+		<>
+			<Masterbar>
+				<Item
+					url={ '#' }
```

Comment:
No need to evaluate the prop if it's a simple string:

```suggestion
					url="#"
```

At the same time, do we really want to have a `#` as a URL, and handle the navigation with `onClick` handlers? Feels a bit counterintuitive and inaccessible - what if I don't click but use the keyboard to navigate?

Reply:
I actually am reusing code from masterbar, but you are right that this is a bit fishy will dig deeper here! :)